### DISCUSSION TITLE
Why is the scrollbar suddenly always displayed.

### DISCUSSION THREAD

**Discussion by MarkCallow** [macvim-dev/macvim#1242] [Q&A] [answered]
I'm using "8.2.3455 (172)" which is the latest.

The scroll bar is always displayed, if there are more lines than fit in the windows. This behavior changed recently. Before the scrollbar appeared only if I started scrolling with 2 fingers on the trackpad. I preferred the previous behavior because the scrollbar takes up space in the window and I have the default window width set to hold 80 characters. Having just discovered _Edit→Global Settings→Toggle XXX Scrollbar_ I am now wondering if the previous behavior was simply no scrollbars and I somehow accidentally toggled the right scrollbar and am misremembering the previous behavior. Can the scroll bar be toggled by a command?

My macOS scroll bar preference is still set to automatic so nothing changed there.

**Comment by eirnym** [comment]
Hi,

The option may be altered from default by your vimrc or a plugin you use.

Try to check `guioptions` setting in runtime and set appropriate setting. Probably, important key is not there. 
Also try to run MacVim without any configuration and check default value

**Reply by MarkCallow** [reply]
@eirnym `guioptions` has 'r' set which explains the scrollbar. However I can't see what is setting it. There is no setting of `guioptions` in `/Applications/MacVim.app/Contents/Resources/vim/{g,}vimrc` or `~/.vim/vimrc`. I have no `~/.vim/gvimrc` or `~/{g,}vimrc`.

**Reply by ychin** [reply]
That's just the builtin default. It's set in the Vim binary itself.

**Comment by ychin** [comment, marked as answer]
I don't think anything has changed. This is just how Vim scrollbars work. It's not easy to just add auto-hiding of scrollbars to MacVim, because the way that the macOS API works isn't just something you can turn on. It also wouldn't work very well since MacVim is kind of weird with its scroll bars with one at the bottom, and multiple ones on the right and on the left. To fix it properly we should just do an auto-hiding scrollbar per window.

I do have some work that will be submitted for Vim as a pull request soon to add auto-hiding scrollbars, so hopefully it will be in MacVim soon.

**Reply by MarkCallow** [reply]
Thank you for the information.
